consensus/safety-rules/src/safety_rules.rs

```diff
@@ -98,6 +106,32 @@ impl<T: Payload> SafetyRules<T> {
         }
     }
 
+    /// This reconciles the key pair of a validator signer with a given validator set
+    /// during epoch changes.
+    fn reconcile_key(&mut self, epoch_state: &EpochState) -> Result<(), Error> {
+        let validator_signer = self.signer()?;
+        if let Some(expected_key) = epoch_state
+            .verifier
+            .get_public_key(&validator_signer.author())
+        {
+            let curr_key = validator_signer.public_key();
+            if curr_key != expected_key {
+                let consensus_key = self
+                    .persistent_storage
+                    .consensus_key_for_version(expected_key)
+                    .expect("Unable to retrieve consensus private key");
+                debug!("Reconcile curr key {} using {}", curr_key, consensus_key);
```

Comment:
fortunately we elide our private keys, but in general we should never print private keys since logs can lead to compromises of private data :/
probably good enough to say Found public key in validator set and call it that.